nova/virt/vmwareapi/vm_util.py

```diff
@@ -870,17 +870,26 @@ def get_vmdk_detach_config_spec(client_factory, device,
     return config_spec
 
 
-def get_vm_extra_config_spec(client_factory, extra_opts):
-    """Builds extra spec fields from a dictionary."""
-    config_spec = client_factory.create('ns0:VirtualMachineConfigSpec')
+def get_vm_extra_config_spec(client_factory, extra_opts, config_spec=None):
```

Comment:
I am not too happy about that function (even originally). It is called `get_vm_extra_config_spec` but returned a `VirtualMachineConfigSpec` with `extraConfig` set.
Now you pass a `config_spec`, because you just want the `extraConfig`, which shows in my opinion, that we need a function which does only that: Creating an `extraConfig`.

My preferred way to do that, would be to extract an function `create_extra_config` and use that function in `get_vm_extra_config_spec`.
I would also vote for renaming `get_vm_extra_config_spec` into `get_vm_config_spec_with_extra_config`, or drop it, and create the config-spec on the caller site.

Reply:
I am not fixed on the `create_` prefix...